File: Omkar's Pet Cat/gemini2.py
def solve():
    import sys
    input = sys.stdin.read
    data = input().split()
    if not data: return
    t = int(data[0])
    idx = 1
    
    out = []
    for _ in range(t):
        n = int(data[idx])
        a = [int(x) for x in data[idx+1 : idx+1+n]]
        idx += n + 1
        
        mn = min(a)
        if a.count(mn) == 1:
            out.append("YES")
        else:
            if any(x % mn != 0 for x in a):
                out.append("YES")
            else:
                out.append("NO")
                
    print('\n'.join(out))

if __name__ == '__main__':
    solve()

File: Omkar's Pet Cat/gemini2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    cin.tie(0)->sync_with_stdio(0);
    int t;
    if (!(cin >> t)) return 0;
    while (t--) {
        int n; 
        cin >> n;
        vector<int> a(n);
        int mn = 2e9;
        for (int& x : a) {
            cin >> x;
            mn = min(mn, x);
        }
        
        int c = count(a.begin(), a.end(), mn);
        if (c == 1) {
            cout << "YES\n";
            continue;
        }
        
        bool ok = false;
        for (int x : a) {
            if (x % mn != 0) {
                ok = true;
                break;
            }
        }
        cout << (ok ? "YES\n" : "NO\n");
    }
    return 0;
}